src/compiler/diagnosticMessages.json

```diff
@@ -1364,6 +1364,26 @@
         "category": "Error",
         "code": 1433
     },
+    "'{0}' is a type and must be imported with a type-only import when 'importsNotUsedAsValues' is set to 'preserve-exact'.": {
+        "category": "Error",
+        "code": 1434
+    },
+    "'{0}' is a type and must be re-exported with a type-only re-export when 'importsNotUsedAsValues' is set to 'preserve-exact'.": {
+        "category": "Error",
+        "code": 1435
+    },
+    "'{0}' resolves to a type-only declaration and must be imported with a type-only import when 'importsNotUsedAsValues' is set to 'preserve-exact'.": {
+        "category": "Error",
+        "code": 1436
+    },
+    "'{0}' resolves to a type-only declaration and must be re-exported with a type-only re-export when 'importsNotUsedAsValues' is set to 'preserve-exact'.": {
+        "category": "Error",
+        "code": 1437
+    },
+    "'{0}' resolves to a type-only declaration and must be re-exported with a type-only re-export when 'isolatedModules' is enabled.": {

```

Comment:
The addition of this diagnostic for `isolatedModules` is actually a bug fix which should have been included in type-only import work from the beginning.